### ISSUE TITLE
Predict Proba return NAN

### ISSUE THREAD

**Issue by beavisluck** [manujosephv/pytorch_tabular#386]
**Describe the bug**
After tabular_model.predict(test) , all of the class probability return nan, and all predictions returns 0, as shown on screenshot below

**To Reproduce**
import pandas as pd
from sklearn.model_selection import train_test_split
from pytorch_tabular import TabularModel
from pytorch_tabular.models import (
    TabTransformerConfig
)
from pytorch_tabular.config import (
    DataConfig,
    OptimizerConfig,
    TrainerConfig,
    ExperimentConfig,
)
from pytorch_tabular.models.common.heads import LinearHeadConfig

df=pd.read_excel('dataset.xlsx',engine='openpyxl')

df=df.fillna(-1)
df=df.drop(df.columns[[0]], axis=1)
X=df.loc[:, df.columns != 'reported']
X=X.drop(['dropped columns'], axis=1)
y=df['flag column']
x_train,x_test, y_train, y_test = train_test_split(X, y,stratify=y, test_size=0.20)
x_train_final,x_val,y_train_final,y_val=train_test_split(x_train, y_train,stratify=y_train, test_size=0.15)
x_train_final['flag']=y_train_final
x_test['flag']=y_test
x_val['flag']=y_val

cat_col=['list of cat col']
num_col=[item for item in list(X.columns) if item not in cat_col]

data_config = DataConfig(
    target=[
        'flag'
    ],  # target should always be a list. Multi-targets are only supported for regression. Multi-Task Classification is not implemented
    continuous_cols=num_col,
    categorical_cols=cat_col
)
trainer_config = TrainerConfig(
    batch_size=1024,
    max_epochs=100,
)
optimizer_config = OptimizerConfig()
model_config = TabTransformerConfig(
    task="classification",
)
tabular_model = TabularModel(
    data_config=data_config,
    model_config=model_config,
    optimizer_config=optimizer_config,
    trainer_config=trainer_config,
    verbose=True
)

tabular_model.fit(train=x_train_final, validation=x_val)

test_feature=x_test.loc[:, x_test.columns != 'flag']
pred_df = tabular_model.predict(test_feature)
pred_df.head()

**Expected behavior**
Supposed to return prediction and predict probability, which shouldn't be zero

**Screenshots**
![image](https://github.com/manujosephv/pytorch_tabular/assets/38403822/eef67e3a-0069-4fc5-883a-640c57e1ba59)

**Desktop (please complete the following information):**

Server

**Smartphone (please complete the following information):**

- Device: \[e.g. iPhone6\]
- OS: \[e.g. iOS8.1\]
- Browser \[e.g. stock browser, safari\]
- Version \[e.g. 22\]

**Additional context**
the training return validation result tho
![image](https://github.com/manujosephv/pytorch_tabular/assets/38403822/adf0ddfe-f5b9-42a4-ac78-124e066b5c74)

**Comment by beavisluck** [comment]
there's nan data on predict dataset, my bad, thanks
